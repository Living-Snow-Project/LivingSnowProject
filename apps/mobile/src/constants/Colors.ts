const tintColor = "#2f95dc";

export const Colors = {
  tintColor,
  tabIconDefault: "#0077ff",
  tabIconSelected: tintColor,
  tabBar: "#fefefe",
  errorBackground: "red",
  errorText: "#fff",
  warningBackground: "#EAEB5E",
  warningText: "#666804",
  noticeBackground: tintColor,
  noticeText: "#fff",
};
