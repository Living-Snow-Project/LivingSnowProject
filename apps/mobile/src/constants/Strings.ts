// for TextInput placeholder prop
export const Placeholders = {
  GPS: {
    AcquiringLocation: "Looking for GPS signal",
    EnterCoordinates: "ie. 12.34567, -123.45678",
    NoPermissions:
      "Check location permissions in Settings. Enter coordinates manually.",
    NoLocation: "Could not determine location. Enter coordinates manually.",
  },
  Settings: {
    Username: "Enter your name",
    Organization: "Enter organization you belong to (if any)",
  },
  RecordScreen: {
    LocationDescription: "ie: Blue Lake, North Cascades, WA",
    Notes: "ie. dark red algae on glacial ice",
    Size: "Select a size",
    TubeId: "Leave blank if tube does not have an id",
    TubeIdDisabled: "Disabled - only used for Samples",
  },
};

export const Labels = {
  Date: "Date",
  Delete: "Delete",
  RecordForm: {
    Color: "Color (select all that apply)",
    Gps: "GPS Coordinates (latitude, longitude)",
    Photos: "Select Photos (limit 4)",
    Type: "Are you Reporting a Sighting or Taking a Sample?",
  },
  // TODO: rename to RecordDetailsFields
  RecordFields: {
    Type: "Type",
    Name: "Name",
    Organization: "Organization",
    TubeId: "Tube Id",
    GPSCoordinates: "Coordinates",
    Latitude: "Latitude",
    Longitude: "Longitude",
    Size: "Size",
    Colors: "Colors",
    LocationDescription: "Location Description",
    Notes: "Notes",
    Photos: "Photos",
  },
  TimelineScreen: {
    ExampleRecords: "Example Record",
    PendingRecords: "Pending",
    DownloadedRecords: "Downloaded",
  },
  SettingsScreen: {
    DarkMode: "Dark",
    ManualCoordinates: "Manual Coordinates Warning",
  },
  StatusBar: {
    NoConnection: "No Internet Connection",
  },
  Modal: {
    Cancel: "Cancel",
    Confirm: "Confirm",
    DiskUsage: {
      header: "Confirm Delete",
      body: "Are you sure you want to delete saved photos?",
    },
    GpsManualEntry: {
      header: "Enter coordinates manually?",
      body: `Some users enter coordinates manually if they cannot acquire GPS signal or when they return home.\n\nThis message can be disabled in Settings.`,
    },
  },
};

// for Modal alerts
export const Notifications = {
  uploadSuccess: {
    title: "Record Uploaded",
    message: "Thanks for your submission.",
  },
  uploadRecordFailed: {
    title: "Record Saved",
    message: "We will upload it later.",
  },
  uploadPhotoFailed: {
    title: "Photo Saved",
    message: "We will upload it later.",
  },
  uploadPhotosFailed: {
    title: "Photos Saved",
    message: "We will upload them later.",
  },
  updateRecordSuccess: {
    title: "Record Updated",
  },
  updateRecordFailed: {
    title: "Update Failed",
    message:
      "This can happen when Internet connectivity is restored while updating the record.",
  },
  backgroundTasksNotAllowed: {
    title: "Background Uploads Not Enabled",
    message:
      "Please enable Background App Refresh if you would like stored records to be uploaded in the background.",
  },
};

export const Validations = {
  invalidCoordinates:
    'Coordinates must be in "lat, long" format. eg. 12.34567, -123.45678',
  invalidAlgaeSize: "A size must be specified",
  invalidAlgaeColors: "At least 1 color must be specified",
};

export const Errors = {
  recordNotFound: "Record to update not found",
};

export const RecordDescription = {
  Sample: `Sample`,
  Sighting: `Sighting`,
  Undefined: "Undefined",
};

export const AlgaeSizeDescription = {
  Select: "Select a size",
  Fist: "Fist",
  ShoeBox: "Shoe Box",
  CoffeeTable: "Coffee Table",
  Car: "Car",
  Bus: "Bus",
  Playground: "Playground",
  SportsField: "Sports Field",
  Other: "Other (describe in notes)",
};

export const AlgaeColorDescription = {
  Select: "Select a color",
  Red: "Red",
  Pink: "Pink",
  Grey: "Grey",
  Green: "Green",
  Orange: "Orange",
  Yellow: "Yellow",
  Other: "Other (describe in notes)",
};

export const BackgroundTasks = {
  UploadData: "background-upload-task",
};

// TODO: move this to Labels
export const Headers = {
  DiskUsage: "Disk Usage",
  Profile: "Profile",
  Prompts: "Prompts",
  Theme: "Theme",
};
