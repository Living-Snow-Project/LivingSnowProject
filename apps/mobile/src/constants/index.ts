export * from "./Colors";
export * from "./Strings";
export * from "./TestIds";
