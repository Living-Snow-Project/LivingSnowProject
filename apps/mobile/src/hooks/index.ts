export * from "./useAlgaeRecords";
export * from "./useCachedPhotos";
export * from "./useCachedResources";
export * from "./useRecordList";
export * from "./useToast";
