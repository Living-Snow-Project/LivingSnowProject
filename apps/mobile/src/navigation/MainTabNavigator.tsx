import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { NavigationContainer } from "@react-navigation/native";
import React from "react";
import TimelineScreen from "../screens/TimelineScreen";
import RecordScreen from "../screens/RecordScreen";
import SettingsScreen from "../screens/SettingsScreen";
import FirstRunScreen from "../screens/FirstRunScreen";
import ImagesPickerScreen from "../screens/ImagesPickerScreen";
import HeaderButton from "../components/HeaderButton";
import RecordDetailsScreen from "../screens/RecordDetailsScreen";
import { RootStackParamList, RootStackNavigationProp } from "./Routes";
import { getAppSettings } from "../../AppSettings";
import TestIds from "../constants/TestIds";

const Stack = createNativeStackNavigator<RootStackParamList>();

type SettingsButtonProps = {
  navigation: RootStackNavigationProp;
};

function SettingsButton({ navigation }: SettingsButtonProps) {
  return (
    <HeaderButton
      testID={TestIds.TimelineScreen.SettingsButton}
      onPress={() => navigation.navigate("Settings")}
      iconName="settings-outline"
      placement="left"
    />
  );
}

type NewRecordButtonProps = {
  navigation: RootStackNavigationProp;
};

function NewRecordButton({ navigation }: NewRecordButtonProps) {
  return (
    <HeaderButton
      testID={TestIds.TimelineScreen.NewRecordButton}
      onPress={() => navigation.navigate("Record")}
      iconName="add-circle-outline"
      placement="right"
    />
  );
}

function RootNavigator() {
  const { showFirstRun } = getAppSettings();

  return (
    <Stack.Navigator
      screenOptions={{ headerShown: true, headerTitleAlign: "center" }}
    >
      {showFirstRun && (
        <Stack.Screen name="Welcome" component={FirstRunScreen} />
      )}
      <Stack.Screen
        name="Timeline"
        component={TimelineScreen}
        options={({ navigation }: { navigation: RootStackNavigationProp }) => ({
          headerLeft: () => SettingsButton({ navigation }),
          headerRight: () => NewRecordButton({ navigation }),
        })}
      />
      <Stack.Screen name="Record" component={RecordScreen} />
      <Stack.Screen name="Settings" component={SettingsScreen} />
      <Stack.Screen
        name="ImageSelection"
        component={ImagesPickerScreen}
        options={{ title: "Camera Roll" }}
      />
      <Stack.Screen
        name="RecordDetails"
        component={RecordDetailsScreen}
        options={{ title: "Details" }}
      />
    </Stack.Navigator>
  );
}

export default function Navigation() {
  return (
    <NavigationContainer>
      <RootNavigator />
    </NavigationContainer>
  );
}

export { SettingsButton, NewRecordButton };
