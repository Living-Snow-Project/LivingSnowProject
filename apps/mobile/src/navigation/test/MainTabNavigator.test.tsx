import React from "react";
import { render, waitFor } from "@testing-library/react-native";
import { NativeBaseProviderForTesting } from "../../../jesttest.setup";
import { Navigation } from "../MainTabNavigator";
import { setAppSettings } from "../../../AppSettings";
import { Labels } from "../../constants/Strings";
import { AlgaeRecordsContext } from "../../hooks/useAlgaeRecords";
import { makeAlgaeRecordsMock } from "../../mocks/useAlgaeRecords.mock";

function WrappedNavigation() {
  return (
    <NativeBaseProviderForTesting>
      <Navigation />
    </NativeBaseProviderForTesting>
  );
}

describe("Navigation test suite", () => {
  test("renders first run screen", () => {
    const { toJSON } = render(<WrappedNavigation />);

    expect(toJSON()).toMatchSnapshot();
  });

  test("renders timeline screen", async () => {
    setAppSettings((prev) => ({ ...prev, showFirstRun: false }));
    const algaeRecords = makeAlgaeRecordsMock({
      isEmpty: true,
    });

    const { getByText, toJSON } = render(
      <AlgaeRecordsContext.Provider value={algaeRecords}>
        <Navigation />
      </AlgaeRecordsContext.Provider>
    );

    await waitFor(() => getByText(Labels.TimelineScreen.ExampleRecords));
    expect(toJSON()).toMatchSnapshot();
  });
});
