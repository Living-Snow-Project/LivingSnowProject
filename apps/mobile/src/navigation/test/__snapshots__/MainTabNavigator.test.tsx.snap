// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation test suite renders first run screen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          Object {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="#0891b2"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="Welcome"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            Object {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "marginTop": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <Text />
                <View
                  dataSet={Object {}}
                  style={Object {}}
                />
                <Text />
                <View
                  dataSet={Object {}}
                  style={Object {}}
                />
                <View
                  dataSet={Object {}}
                  style={Object {}}
                >
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#db2777",
                        "fontFamily": undefined,
                        "fontSize": 20,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 30,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    Living Snow Project
                  </Text>
                  <View
                    dataSet={Object {}}
                    style={Object {}}
                  />
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#ec4899",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    See pink snow? Let us know!
                  </Text>
                </View>
                <View
                  dataSet={Object {}}
                  style={Object {}}
                />
                <Text />
                <View
                  dataSet={Object {}}
                  style={Object {}}
                />
                <Text />
              </View>
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "marginBottom": 8,
                    "marginTop": 8,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Let us know who you are or remain anonymous. You can update your profile at any time in the Settings tab.
              </Text>
              <View
                dataSet={Object {}}
                isInvalid={false}
                isRequired={false}
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  feedbackId="field-3-feedback"
                  hasFeedbackText={true}
                  hasHelpText={false}
                  helpTextId="field-3-helptext"
                  isDisabled={false}
                  isInvalid={false}
                  isReadOnly={false}
                  isRequired={false}
                  labelId="field-3-label"
                  nativeID="field-3-label"
                  setHasFeedbackText={[Function]}
                  setHasHelpText={[Function]}
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "marginBottom": 4,
                      "marginTop": 4,
                    }
                  }
                >
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#737373",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "500",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    Name
                  </Text>
                </View>
                <View
                  dataSet={Object {}}
                  isFocused={false}
                  style={
                    Object {
                      "alignItems": "center",
                      "borderColor": "#d4d4d4",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "overflow": "hidden",
                    }
                  }
                >
                  <TextInput
                    accessibilityDescribedBy="field-3-feedback"
                    accessible={true}
                    blurOnSubmit={false}
                    dataSet={Object {}}
                    disabled={false}
                    editable={true}
                    isRequired={false}
                    maxLength={50}
                    nativeID="field-3-input"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Enter your name"
                    placeholderTextColor="#a3a3a3"
                    readOnly={false}
                    required={false}
                    returnKeyType="done"
                    secureTextEntry={false}
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#171717",
                        "flex": 1,
                        "fontFamily": undefined,
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "height": "100%",
                        "paddingBottom": 8,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 8,
                        "width": "100%",
                      }
                    }
                  />
                </View>
              </View>
              <View
                dataSet={Object {}}
                isInvalid={false}
                isRequired={false}
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  feedbackId="field-4-feedback"
                  hasFeedbackText={true}
                  hasHelpText={false}
                  helpTextId="field-4-helptext"
                  isDisabled={false}
                  isInvalid={false}
                  isReadOnly={false}
                  isRequired={false}
                  labelId="field-4-label"
                  nativeID="field-4-label"
                  setHasFeedbackText={[Function]}
                  setHasHelpText={[Function]}
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "marginBottom": 4,
                      "marginTop": 4,
                    }
                  }
                >
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#737373",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "500",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    Organization
                  </Text>
                </View>
                <View
                  dataSet={Object {}}
                  isFocused={false}
                  style={
                    Object {
                      "alignItems": "center",
                      "borderColor": "#d4d4d4",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "overflow": "hidden",
                    }
                  }
                >
                  <TextInput
                    accessibilityDescribedBy="field-4-feedback"
                    accessible={true}
                    blurOnSubmit={true}
                    dataSet={Object {}}
                    disabled={false}
                    editable={true}
                    isRequired={false}
                    maxLength={50}
                    nativeID="field-4-input"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onKeyPress={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Enter organization you belong to (if any)"
                    placeholderTextColor="#a3a3a3"
                    readOnly={false}
                    required={false}
                    returnKeyType="done"
                    secureTextEntry={false}
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#171717",
                        "flex": 1,
                        "fontFamily": undefined,
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "height": "100%",
                        "paddingBottom": 8,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 8,
                        "width": "100%",
                      }
                    }
                  />
                </View>
              </View>
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessible={true}
                  collapsable={false}
                  colorScheme="primary"
                  dataSet={Object {}}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#0891b2",
                      "borderRadius": 4,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginTop": 20,
                      "paddingBottom": 10,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 10,
                      "width": 200,
                    }
                  }
                >
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                    test={true}
                  >
                    <View
                      dataSet={Object {}}
                      style={Object {}}
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#fafafa",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        Start Reporting
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Navigation test suite renders timeline screen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          Object {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="#0891b2"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="Timeline"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        >
          <RNSScreenStackHeaderSubview
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            type="left"
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "marginLeft": 25,
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
              testID="timeline-screen-settings-button"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#0891b2",
                      "fontSize": 32,
                    },
                    undefined,
                    Object {
                      "fontFamily": "ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="timeline-screen-settings-button-settings-outline"
              >
                
              </Text>
            </View>
          </RNSScreenStackHeaderSubview>
          <RNSScreenStackHeaderSubview
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            type="right"
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "marginRight": 25,
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
              testID="timeline-screen-record-button"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#0891b2",
                      "fontSize": 32,
                    },
                    undefined,
                    Object {
                      "fontFamily": "ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="timeline-screen-record-button-add-circle-outline"
              >
                
              </Text>
            </View>
          </RNSScreenStackHeaderSubview>
        </RNSScreenStackHeaderConfig>
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            Object {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgba(214, 211, 209, 1)",
                }
              }
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textAlign": "center",
                    "textDecorationLine": undefined,
                  }
                }
              >
                No Internet Connection
              </Text>
            </View>
            <RCTScrollView
              ListEmptyComponent={[Function]}
              contentContainerStyle={
                Array [
                  Object {},
                  Object {
                    "dataSet": Object {},
                  },
                  Object {},
                ]
              }
              data={Array []}
              dataSet={Object {}}
              getItem={[Function]}
              getItemCount={[Function]}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onEndReached={[Function]}
              onEndReachedThreshold={5}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onRefresh={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              refreshControl={
                <RefreshControlMock
                  onRefresh={[Function]}
                  refreshing={false}
                />
              }
              refreshing={false}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              style={Object {}}
              testID="timeline-screen-flatlist"
              viewabilityConfigCallbackPairs={Array []}
            >
              <RCTRefreshControl />
              <View>
                <View
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": "#a8a29e",
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textAlign": "center",
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    Example Record
                  </Text>
                </View>
                <View
                  accessible={true}
                  collapsable={false}
                  dataSet={Object {}}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={Object {}}
                  testID="1234"
                >
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "paddingBottom": 4,
                        "paddingLeft": 8,
                        "paddingRight": 8,
                        "paddingTop": 4,
                      }
                    }
                  >
                    <View
                      dataSet={Object {}}
                      style={Object {}}
                    >
                      <View
                        dataSet={Object {}}
                        style={
                          Object {
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          dataSet={Object {}}
                          style={
                            Object {
                              "width": "15%",
                            }
                          }
                        >
                          <View
                            dataSet={Object {}}
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#f0abfc",
                                "borderColor": "#27272a",
                                "borderRadius": 9999,
                                "height": 48,
                                "justifyContent": "center",
                                "marginTop": 4,
                                "position": "relative",
                                "width": 48,
                              }
                            }
                          >
                            <Text
                              dataSet={Object {}}
                              style={
                                Object {
                                  "backgroundColor": undefined,
                                  "color": "#fafafa",
                                  "fontFamily": undefined,
                                  "fontSize": 16,
                                  "fontStyle": "normal",
                                  "fontWeight": "600",
                                  "letterSpacing": 0,
                                  "lineHeight": 24,
                                  "textDecorationLine": undefined,
                                }
                              }
                            >
                              HI
                            </Text>
                          </View>
                        </View>
                        <View
                          dataSet={Object {}}
                          style={
                            Object {
                              "paddingRight": 4,
                              "width": "78%",
                            }
                          }
                        >
                          <View
                            dataSet={Object {}}
                            style={
                              Object {
                                "marginLeft": 8,
                              }
                            }
                          >
                            <Text
                              dataSet={Object {}}
                              style={
                                Object {
                                  "backgroundColor": undefined,
                                  "color": "#171717",
                                  "fontFamily": undefined,
                                  "fontSize": 14,
                                  "fontStyle": "normal",
                                  "fontWeight": "600",
                                  "letterSpacing": 0,
                                  "lineHeight": 21,
                                  "textDecorationLine": undefined,
                                }
                              }
                            >
                              Helpful Individual
                            </Text>
                            <View
                              dataSet={Object {}}
                              style={Object {}}
                            />
                            <Text
                              dataSet={Object {}}
                              style={
                                Object {
                                  "backgroundColor": undefined,
                                  "color": "#171717",
                                  "fontFamily": undefined,
                                  "fontSize": 14,
                                  "fontStyle": "normal",
                                  "fontWeight": "400",
                                  "letterSpacing": 0,
                                  "lineHeight": 21,
                                  "textDecorationLine": undefined,
                                }
                              }
                            >
                              Community Scientist
                            </Text>
                            <View
                              dataSet={Object {}}
                              style={Object {}}
                            />
                            <Text
                              dataSet={Object {}}
                              style={
                                Object {
                                  "backgroundColor": undefined,
                                  "color": "#db2777",
                                  "fontFamily": undefined,
                                  "fontSize": 14,
                                  "fontStyle": "normal",
                                  "fontWeight": "700",
                                  "letterSpacing": 0,
                                  "lineHeight": 21,
                                  "textDecorationLine": undefined,
                                }
                              }
                            >
                              Sample, 2022-04-01
                            </Text>
                          </View>
                        </View>
                        <View
                          dataSet={Object {}}
                          style={Object {}}
                        />
                        <View
                          dataSet={Object {}}
                          style={
                            Object {
                              "width": "7%",
                            }
                          }
                        />
                      </View>
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      />
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "500",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        White Chuck Glacier, Glacier Peak Wilderness, WA
                      </Text>
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      />
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        Dark red algae in runnels
                      </Text>
                    </View>
                  </View>
                  <Image
                    accessibilityLabel="static image"
                    alt="static image"
                    dataSet={Object {}}
                    onError={[Function]}
                    source={1}
                    style={
                      Object {
                        "maxWidth": "100%",
                      }
                    }
                  />
                </View>
                <View
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": "#d6d3d1",
                      "height": 12,
                      "width": "100%",
                    }
                  }
                />
              </View>
            </RCTScrollView>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
